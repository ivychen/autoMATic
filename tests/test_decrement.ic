/* autoMATic Compiler - Spring 2018 - COMS4115 PLT
 by Ivy Chen ic2389, Nguyen Chi Dung ncd2118,
 Nelson Gomez ng2573, Jimmy O'Donnell jo2474 */

int main() {
	int i;
	int j;

	i = 5;
	j = i;

	println(i); // 5
  println(j); // 5

  i--;
  j--;

  j = j - 1;
  if (i-- == j) println("i-- == j"); // true
  j = j - 1;
  if (i-- == j) println("i-- == j"); // true
  j = j - 1;
  if (i-- == j) println("i-- == j"); // true

  println(i); // 1
  println(j); // 1

	return 0;
}
