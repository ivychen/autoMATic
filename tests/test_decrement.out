5
5
i-- == j
i-- == j
i-- == j
1
1
